src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/ECDsa/ECDsaTests.cs

```diff
@@ -2,18 +2,201 @@
+                        }
+                    }
+                    else if (param.Curve.IsCharacteristic2)
+                    {
+                        if (curveDef.Curve.Oid.Value != ECDSA_Sect193r1_OID_VALUE)
+                        {
+                            if (ECDsaFactory.IsCurveValid(new Oid(ECDSA_Sect193r1_OID_VALUE)))
+                            {
+                                // Specify different curve by name
+                                newCng.GenerateKey(ECCurve.CreateFromValue(ECDSA_Sect193r1_OID_VALUE));
+                                Assert.Equal(193, newCng.KeySize);
+                                param2 = newCng.ExportExplicitParameters(true);
+                                // Keys should should not match
+                                ComparePrivateKey(param, param2, false);
+                                ComparePublicKey(param.Q, param2.Q, false);
+                                // Polynomial,X,Y (and others) should not match
+                                Assert.True(param2.Curve.IsCharacteristic2);
+                                Assert.NotEqual(param.Curve.Polynomial, param2.Curve.Polynomial);
+                                Assert.NotEqual(param.Curve.G.X, param2.Curve.G.X);
+                                Assert.NotEqual(param.Curve.G.Y, param2.Curve.G.Y);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        [Theory, MemberData(nameof(TestCurves))]
+        public static void TestRegenKeyNamed(CurveDef curveDef)
+        {
+            ECParameters param, param2;
+            ECDsa cng;
+
+            using (cng = ECDsaFactory.Create(curveDef.Curve))
+            {
+                param = cng.ExportParameters(true);
+                Assert.NotEqual(param.D, null);
+                param.Validate();
+
+                cng.GenerateKey(param.Curve);
+                param2 = cng.ExportParameters(true);
+                param2.Validate();
+
+                // Only curve should match
+                ComparePrivateKey(param, param2, false);
+                ComparePublicKey(param.Q, param2.Q, false);
+                CompareCurve(param.Curve, param2.Curve);
+            }
+        }
+
+        [Theory]
+        public static void TestRegenKeyNistP256()
+        {
+            ECParameters param, param2;
+            ECDsa cng;
+
+            using (cng = ECDsaFactory.Create(256))
+            {
+                param = cng.ExportExplicitParameters(true);
+                Assert.NotEqual(param.D, null);
+
+                cng.GenerateKey(param.Curve);
+                param2 = cng.ExportExplicitParameters(true);
+
+                // Only curve should match
+                ComparePrivateKey(param, param2, false);
+                ComparePublicKey(param.Q, param2.Q, false);
+                CompareCurve(param.Curve, param2.Curve);
+            }
+        }
+
```

Comment:
I added a test that does that. Compares curve name and algorithm (e.g. CngAlgorithm.ECDsaP256)